bokeh/models/sources.py

```diff
@@ -38,6 +39,10 @@ class DataSource(PlotObject):
                 hit/selected
     """)
 
+    callback = Instance(Callback, help="""
+    A callback to run in the browser whenever the selection is changed.
+    """)
+
```

Comment:
Well, in general there is no way to know if someone "reaches inside" a column and changes it in place. This is the same problem the Traits library had with numpy arrays. We can make an interface that wraps that operation so that there is a hook we could use, but there is no recourse if someone bypasses that interface and updates data directly.